students/Kristy_Martini/lesson09/mailroom.py

```diff
@@ -0,0 +1,149 @@
+import operator
+import os
+
+class Report:
+    def __init__(self):
+        """ Report class holds the list of all donors and information associated with them."""
+        self.donors = dict()
+        Kristy = Donor("Kristy Martini", 0, 3, 3000)
+        Mikey = Donor("Mike Martini", 0, 7, 3424834)
+        Cathy = Donor("Cathy Martini", 0, 7, 63833)
+        Bill = Donor("Bill Martini", 0, 10, 60000)
+        Nick = Donor("Nick Martini", 0, 50, 47484949)
+
+        initial_donors = [Kristy, Mikey, Nick, Bill, Cathy]
+        [self.add_donor(donor) for donor in initial_donors]
+
+    def add_donor(self, Donor):
+        """ Add new donor to report"""
+        self.donors[Donor.name] = Donor
+    
+    def sort_donors(self, new_report=None):
+        """ Sorts donors by total gift value given"""
+        if new_report is None:
+            new_report = self
+        new_report.unsorted_list = list(new_report.donors.values())
+        new_report.sorted_list = sorted(new_report.unsorted_list, key=operator.attrgetter('total_gift_value'), reverse=True)
+        new_report.sorted_dict = dict()
+        for donor in new_report.sorted_list:
+            new_report.sorted_dict[donor.name] = donor
+
+    def create_report(self, new_report=None):
+        """ Display donor report to the user"""
+        if new_report is None:
+            new_report = Report()
+        new_report.sort_donors(new_report)
+        report_lines = []
+        report_lines.append("Donor Name          | Total Given   | Num Gifts | Average Gift")
+        report_lines.append("--------------------------------------------------------------")
+        for value in new_report.sorted_dict.values():
+            new_line = '{0:21}'.format(value.name) + "$" + '{0:14}'.format(value.total_gift_value) + '{0:12}'.format(value.num_gifts) + " $" + '{0:12}'.format(value.average_gift)
+            report_lines.append(new_line)
+        report_lines.append("\n")
+        return report_lines
```

Comment:
above patterns aren't following best practices:

`new_report = self` - your report data is local to the function. Use a local variable and return the sorted data.
if you really want to store the data, create a class attribute `self.sorted_data` in `__init__`

same with `create_report`,  you would not create an instance of itself unless it was a `staticmethod` which you don't need anyways. This function should use variables with local scope -- to access your sort function: `self.sort_donors()`